File: backend/Terena.Services/Database/Venue.cs
using System;
using System.Collections.Generic;

namespace Terena.Services.Database
{
    public class Venue
    {
        public int Id { get; set; }
        public string Name { get; set; }
        public string Location { get; set; }
        public string Address { get; set; }
        public string SportType { get; set; }
        public string SurfaceType { get; set; }
        public decimal PricePerHour { get; set; }
        public int AvailableSlots { get; set; }
        public string Description { get; set; }
        public string ContactPhone { get; set; }
        public string ContactEmail { get; set; }
        public string VenueImageUrl { get; set; }
        
        public bool HasParking { get; set; }
        public bool HasShowers { get; set; }
        public bool HasLighting { get; set; }
        public bool HasChangingRooms { get; set; }
        public bool HasEquipmentRental { get; set; }
        public bool HasCafeBar { get; set; }
        public bool HasWaterFountain { get; set; }
        public bool HasSeatingArea { get; set; }
        
        public List<VenueAmenity> Amenities { get; set; }
        public List<OperatingHour> OperatingHours { get; set; }
        public CancellationPolicy? CancellationPolicy { get; set; }
        public Discount? Discount { get; set; }
        public int? CancellationPolicyHours { get; set; }
        public decimal? DiscountPercentage { get; set; }
        public int? DiscountThreshold { get; set; }
        public decimal? CancellationFeePercentage { get; set; }
        public int? RefundProcessingDays { get; set; }
        public ICollection<Court>? Courts { get; set; }
        public ICollection<Booking>? Bookings { get; set; }
        public ICollection<Review>? Reviews { get; set; }
        public ICollection<UserFavoriteVenue>? FavoritedBy { get; set; }
    }

    public class VenueAmenity
    {
        public int Id { get; set; }
        public string Name { get; set; }
        public bool IsAvailable { get; set; }
        public int VenueId { get; set; }
    }

    public class OperatingHour
    {
        public int Id { get; set; }
        public string Day { get; set; }
        public string StartTime { get; set; }
        public string EndTime { get; set; }
        public int VenueId { get; set; }
        public Venue Venue { get; set; }
    }

    public class CancellationPolicy
    {
        public int Id { get; set; }
        public DateTime FreeUntil { get; set; }
        public decimal Fee { get; set; }
        public int VenueId { get; set; }
        public Venue Venue { get; set; }
    }

    public class Discount
    {
        public int Id { get; set; }
        public decimal Percentage { get; set; }
        public int ForBookings { get; set; }
        public int VenueId { get; set; }
        public Venue Venue { get; set; }
    }
}

File: frontend/terena_mobile/lib/models/venue.dart
class Venue {
  final int id;
  final String name;
  final String? location;
  final String? address;
  final String? imageUrl;
  final double? pricePerHour;
  final double? rating;
  final int? reviewCount;
  final String? sportType;
  final String? surfaceType;
  final String? description;
  final bool hasParking;
  final bool hasShowers;
  final bool hasLighting;
  final bool hasChangingRooms;
  final bool hasEquipmentRental;
  final bool hasCafeBar;
  final Discount? discount;
  final CancellationPolicy? cancellationPolicy;

  Venue({
    required this.id,
    required this.name,
    this.location,
    this.address,
    this.imageUrl,
    this.pricePerHour,
    this.rating,
    this.reviewCount,
    this.sportType,
    this.surfaceType,
    this.description,
    this.hasParking = false,
    this.hasShowers = false,
    this.hasLighting = false,
    this.hasChangingRooms = false,
    this.hasEquipmentRental = false,
    this.hasCafeBar = false,
    this.discount,
    this.cancellationPolicy,
  });

  factory Venue.fromJson(Map<String, dynamic> json) {
    return Venue(
      id: json['id'],
      name: json['name'] ?? '',
      location: json['location'],
      address: json['address'],
      imageUrl: json['venueImageUrl'],
      pricePerHour: json['pricePerHour']?.toDouble(),
      rating: json['averageRating']?.toDouble(),
      reviewCount: json['totalReviews'],
      sportType: json['sportType'],
      surfaceType: json['surfaceType'],
      description: json['description'],
      hasParking: json['hasParking'] ?? false,
      hasShowers: json['hasShowers'] ?? false,
      hasLighting: json['hasLighting'] ?? false,
      hasChangingRooms: json['hasChangingRooms'] ?? false,
      hasEquipmentRental: json['hasEquipmentRental'] ?? false,
      hasCafeBar: json['hasCafeBar'] ?? false,
      discount:
          json['discount'] != null ? Discount.fromJson(json['discount']) : null,
      cancellationPolicy:
          json['cancellationPolicy'] != null
              ? CancellationPolicy.fromJson(json['cancellationPolicy'])
              : null,
    );
  }

  Map<String, dynamic> toJson() {
    return {
      'id': id,
      'name': name,
      'location': location,
      'address': address,
      'venueImageUrl': imageUrl,
      'pricePerHour': pricePerHour,
      'averageRating': rating,
      'totalReviews': reviewCount,
      'sportType': sportType,
      'surfaceType': surfaceType,
      'description': description,
      'hasParking': hasParking,
      'hasShowers': hasShowers,
      'hasLighting': hasLighting,
      'hasChangingRooms': hasChangingRooms,
      'hasEquipmentRental': hasEquipmentRental,
      'hasCafeBar': hasCafeBar,
      'discount': discount?.toJson(),
      'cancellationPolicy': cancellationPolicy?.toJson(),
    };
  }

  String? get venueImageUrl => imageUrl;
}

class Discount {
  final double? percentage;
  final int? forBookings;

  Discount({this.percentage, this.forBookings});

  factory Discount.fromJson(Map<String, dynamic> json) {
    return Discount(
      percentage: json['percentage']?.toDouble(),
      forBookings: json['forBookings'],
    );
  }

  Map<String, dynamic> toJson() {
    return {'percentage': percentage, 'forBookings': forBookings};
  }
}

class CancellationPolicy {
  final DateTime? freeUntil;
  final double? fee;

  CancellationPolicy({this.freeUntil, this.fee});

  factory CancellationPolicy.fromJson(Map<String, dynamic> json) {
    return CancellationPolicy(
      freeUntil:
          json['freeUntil'] != null ? DateTime.parse(json['freeUntil']) : null,
      fee: json['fee']?.toDouble(),
    );
  }

  Map<String, dynamic> toJson() {
    return {'freeUntil': freeUntil?.toIso8601String(), 'fee': fee};
  }
}
